src/extension.ts

```diff
@@ -129,6 +129,15 @@ export async function activate(context: vscode.ExtensionContext) {
 
   const client = new ClangdLanguageClient('Clang Language Server',
                                           serverOptions, clientOptions);
+  client.onDidChangeState(stateChange => {
```

Comment:
As this is only for type-hierarchy, I would use a specific name, and move this part of code to `type-hierarchy.ts` (to make it self-contained, it would be easier for us to delete it once there is an official implementation in VSCode).

As mentioned in another comment, if we can make `hastypeHierarchyProvider` work, we could reuse the `hastypeHierarchyProvider` as well.

Reply:
Refactored as suggested.